### PR TITLE
Add Vala GtkTemplate example

### PR DESCRIPTION
**Pull request by tintou**
There is no official documentation so here it is!

### CODE REVIEW

**Comment by ZanderBrown** [comment]
The compiled `.gresource` probably shouldn't be commited

**Comment by tintou** [comment]
@ZanderBrown oops, removed

File: examples/gtk+-3.0/Gtk.Builder.vala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+	[GtkCallback]
+	private void entry_changed (Gtk.Editable editable) {
+		message ("Entry changed, text: %s\n", entry.text);
+
+		notify_property ("text");
+	}
+}
+
+public static int main (string[] args) {
```
Comment (benwaffle) [inline examples/gtk+-3.0/Gtk.Builder.vala]: can just be `void main (string[] args)` and then get rid of the return statement below

**Comment by tintou** [comment]
@benwaffle done

**Comment by benwaffle** [comment]
`public static` is confusing, since it's not inside an object

**Comment by tintou** [comment]
@benwaffle done

**Comment by benwaffle** [comment]
does a `public` function mean anything outside a class? is `void main(string[] args)` the exact same thing?

**Comment by ZanderBrown** [comment]
@benwaffle if this where a library it would be included in generated vapi/h ext but for a binary it shouldn't make a difference (that i know of)

**Pull request merged**